clustermesh/clustermesh.go

```diff
@@ -2158,3 +2250,59 @@ func updateClustermeshConfig(
 
 	return newValues, nil
 }
+
+func removeFromClustermeshConfig(values map[string]interface{}, aiRemote *accessInformation) (map[string]any, error) {
+	// get current clusters config slice, if it exists
+	c, found, err := unstructured.NestedFieldCopy(values, "clustermesh", "config", "clusters")
+	if err != nil {
+		return nil, fmt.Errorf("existing clustermesh.config is invalid")
+	}
+	if found && c == nil {
```

Comment:
```suggestion
	if !found || c == nil {
```

I guess (dropping the `if !found` below).

Reply:
I think this is correct. Field copy could return found=true with a nil copy.

Reply:
Yeah, I mean that regardless of whether it is found but the value is nil, or it is not found, I guess that you could return an error here.

Reply:
Decided to axe the validation and simply just gen the config. This should alleviate the issue you'd pointed out where one cluster is left dangling.